lib/src/realm_class.dart

```diff
@@ -95,16 +95,45 @@ class Realm implements Finalizable {
   /// Opens a `Realm` using a [Configuration] object.
   Realm(Configuration config) : this._(config);
 
-  Realm._(this.config, [RealmHandle? handle]) : _handle = handle ?? _openRealm(config) {
+  Realm._(this.config, [RealmHandle? handle]) : _handle = handle ?? _openRealmSync(config) {
     _populateMetadata();
   }
 
-  static RealmHandle _openRealm(Configuration config) {
-    var dir = File(config.path).parent;
+  /// A method for asynchronously obtaining and opening a [Realm].
+  ///
+  /// If the configuration is [FlexibleSyncConfiguration], the realm will be downloaded and fully
+  /// synchronized with the server prior to the completion of the returned [RealmAsyncOpenTask].
+  /// Otherwise this method will throw an exception.
+  ///
+  /// Open realm async arguments are:
+  /// * `config`- a configuration object that describes the realm.
+  /// * `onProgressCallback` - a function that is registered as a callback for receiving download progress notifications. It is not mandatory.
+  ///
+  /// The returned task has an awaitable [RealmAsyncOpenTask.realm] future that is completed once the remote realm is fully synchronized.
+  /// It provides a method [RealmAsyncOpenTask.cancel] that cancels any current running download.
+  static RealmAsyncOpenTask open(Configuration config, {ProgressCallback? onProgressCallback}) {
+    _createFileDirectory(config.path);
+    RealmAsyncOpenTaskHandle realmAsyncOpenTaskHandle = realmCore.createRealmAsyncOpenTask(config);
+    final completer = Completer<RealmHandle?>();
+    Future<Realm?> realm = realmCore.openRealmAsync(realmAsyncOpenTaskHandle, completer).onError((error, stackTrace) {
+      print(error);
+      return null;
```

Comment:
Return null only when the error is that operation is canceled. Otherwise we may have to return an exception.